test/e2e/test_job_submissions.py

```diff
@@ -1020,24 +1060,30 @@ def test_worker_run_with_number_of_environments(
 
         job.wait_until_complete(client=deadline_client)
 
-        # Retrieve job output and verify whoami printed the queue's jobsRunAsUser
-        job_logs = job.get_logs(
-            deadline_client=deadline_client,
-            logs_client=boto3.client(
-                "logs",
-                config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
-            ),
-        )
-
-        full_log = "\n".join(
-            [le.message for _, log_events in job_logs.logs.items() for le in log_events]
-        )
+        assert job.task_run_status == TaskStatus.SUCCEEDED
 
-        assert full_log.count("Hello!") == len(
-            job_environments
-        ), "Expected number of Hello statements not found in job logs."
+        if len(job_environments) == 1:
+            job.assert_single_task_log_contains(
+                deadline_client=deadline_client,
+                logs_client=boto3.client(
+                    "logs",
+                    config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+                ),
+                # pass in alldot pattern
+                expected_pattern=r"Hello!",
+                assert_fail_msg="Expected Number of Hello statements not found in job logs.",
+            )
 
-        assert job.task_run_status == TaskStatus.SUCCEEDED
+        if len(job_environments) == 3:
+            job.assert_single_task_log_contains(
+                deadline_client=deadline_client,
+                logs_client=boto3.client(
+                    "logs",
+                    config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+                ),
```

Comment:
nit: looks like we already create a logs boto3 client on lines 1068-1071. We could share the same logs client for both calls

Reply:
Updated